api-reference/_hidden/GridBaseColumn/lookup/displayExpr.md

```diff
@@ -22,139 +22,227 @@ This property accepts a string - the name of the data field that provides displa
+        import Dx{WidgetName}, { DxColumn, DxLookup } from 'devextreme-vue/{widget-name}';
+
+        const calculateSortValue = (data) => {
+            const column = this;
+            const value = column.calculateCellValue(data);
+            return column.lookup.calculateCellValue(value);
+        }
+        </script>
+
+    ##### React
+
+        <!-- tab: App.js -->
+        import React from 'react';
+        import 'devextreme/dist/css/dx.light.css';
+
+        import {WidgetName}, {
+            Column,
+            Lookup
+        } from 'devextreme-react/{widget-name}';
+
+        function calculateSortValue (data) {
+            const column = this;
             const value = column.calculateCellValue(data);
             return column.lookup.calculateCellValue(value);
         }
-    }
-
-    <!-- tab: app.module.ts -->
-    import { BrowserModule } from '@angular/platform-browser';
-    import { NgModule } from '@angular/core';
-    import { AppComponent } from './app.component';
-
-    import { Dx{WidgetName}Module } from 'devextreme-angular';
-
-    @NgModule({
-        declarations: [
-            AppComponent
-        ],
-        imports: [
-            BrowserModule,
-            Dx{WidgetName}Module
-        ],
-        providers: [ ],
-        bootstrap: [AppComponent]
-    })
-    export class AppModule { }
-
-##### Vue
-
-    <!-- tab: App.vue -->
-    <template>
-        <Dx{WidgetName} ... >
-            <DxColumn ...
-                :calculate-sort-value="calculateSortValue">
-                <DxLookup ... />
-            </DxColumn>
-        </Dx{WidgetName}>
-    </template>
-
-    <script>
-    import 'devextreme/dist/css/dx.light.css';
-
-    import Dx{WidgetName}, {
-        DxColumn,
-        DxLookup
-    } from 'devextreme-vue/{widget-name}';
-
-    export default {
-        components: {
-            Dx{WidgetName},
-            DxColumn,
-            DxLookup
-        },
-        data() {
-            return {
-                calculateSortValue (data) {
-                    const column = this;
-                    const value = column.calculateCellValue(data);
-                    return column.lookup.calculateCellValue(value);
+
+        export default function App() {
+            return (
+                <{WidgetName} ... >
+                    <Column ...
+                        calculateSortValue={calculateSortValue}>
+                        <Lookup ... />
+                    </Column>
+                </{WidgetName}>
+            );
+        }
+
+    ---
+
+- When [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) and **columns**.[lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/) are enabled at the same time, the component adds a `null` value to lookup datasets. To avoid errors, check that items are not `null` before accessing them in the **displayExpr** method:
```

Comment:
> the component adds a `null` value to lookup datasets
That's not exactly the case. The component adds a null value to the headerFilter datasets, and based on this value, it tries to retrieve the display value from the lookup.